### DISCUSSION TITLE
importing viewer service error

### DISCUSSION THREAD

**Discussion by smoczyna** [stephanrauh/ngx-extended-pdf-viewer#1763] [Q&A] [answered]
I have the following error when trying to use the service provided along with the viewer. Importing it in any way causes this compilation error.

`This likely means that the library (ngx-extended-pdf-viewer/lib/ngx-extended-pdf-viewer.service) which declares NgxExtendedPdfViewerService is not compatible with Angular Ivy. Check if a newer version of the library is available, and update if so. Also consider checking with the library's authors to see if the library is expected to be compatible with Ivy.`

I am using angular 16 and the latest version of the viewer. 
I'm wondering if this is library issue or misconfiguration ?
Also is there any workaround of this problem if it i s library issue?

**Comment by stephanrauh** [comment]
Sounds like a misconfiguration to me, but I don't have a clue what's going wrong. I've migrated the library to Ivy a long time ago. Since version 11, I've even dropped the support for the old view engine. So the error message puzzles me. My library requires Ivy - exactly opposite to what the error message says.

My general advice to solve this class of problems is to create a new project, just to see that everything works. After getting the new project up and running, you can compare the new project to your real project. Usually, this approach helps you to pin down the issue. Creating a new project is a matter of two minutes:
- ng new TestProject
- cd TestProject
- ng add ngx-extended-pdf-viewer (and accept all the defaults)
- replace the content of the `app.component.html` with `<app-example-pdf-viewer></app-example-pdf-viewer>`
- ng s -o

**Reply by smoczyna** [reply]
I have generated a new project with angular 14 with the same error at the end. One thing I cannot do is running ng script. Scripting is forbidden on my company machine so I have to use npm insead, I called "npm install ngx-extended-pdf-viewer and modified angular.json assets section. Is this all I have to do or I messed something ?
I have generated new app on my home machine and it works out of the box indeed so it must be the restriction of the company laptop, something what "ng add ngx-extended-pdf-viewer" does but npm install doesn't.

**Reply by stephanrauh** [reply]
Oh, that's an annoying restriction. I know from your next message you've already puzzled it out, but if you want to know exactly what "ng add ngx-extended-pdf-viewer" does, here's the algorithm: https://github.com/stephanrauh/ngx-extended-pdf-viewer/tree/main/projects/ngx-extended-pdf-viewer/schematics/ng-add (starting with the index.js - the other files are mostly syntactic bloat).

**Comment by smoczyna** [comment, marked as answer]
I think I got this. I have generated new app on my home machine and compared them with pain. I have it working at last
